addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2331,6 +2346,43 @@ export class OdooEditor extends EventTarget {
                 }
             }
         } else if (node.nodeType !== Node.TEXT_NODE) {
+            if (node.nodeName === 'TD' && (node.hasAttribute('style') || node.hasAttribute('bgcolor'))) {
+                // TD tags do not support style in v15, move style to child or new SPAN.
+                const keepChild = (node.childNodes.length === 1 && node.childNodes[0].nodeName === 'SPAN') 
+                const styleElem = keepChild ? node.childNodes[0] : this.document.createElement('SPAN');
+                if (node.hasAttribute('bgcolor') && !styleElem.style.cssText.includes('background-color:')) {
+                    // bgcolor attribute is deprecated, replaced with background-color.
+                    styleElem.setAttribute('style', 
+                        `${styleElem.getAttribute('style') || ''};background-color:${node.getAttribute('bgcolor')}`);
+                }
+                styleElem.setAttribute('style', `${styleElem.getAttribute('style') || ''};${node.getAttribute('style') || ''}`);
+                // Give TD's children their new parent.
+                if (!keepChild) {
+                    [...node.childNodes].forEach(child => styleElem.appendChild(child));
+                    node.appendChild(styleElem);
+                }
+                this._cleanForPaste(styleElem);
+            } else if (node.nodeName === 'FONT') {
+                // FONT tags have some style information in custom attributes,
+                // this maps them to the style attribute.
+                const attributeMap = { 'color': 'color', 'size': 'font-size'};
+                const existingStyles = (node.getAttribute('style') || '').split(';').map(x => x.trim().split(':')[0]);
+                const newStyles = [...node.attributes]
+                    .filter(att => attributeMap[att.name] && !existingStyles.includes(att.name))
+                    .map(att =>`${attributeMap[att.name]}:${att.value}`).join(';');
+                if (newStyles) {
+                    node.setAttribute('style', `${node.getAttribute('style') || ''};${newStyles}`);
+                }
```

Comment:
Unless I missed something, you don't have to do all that parsing and stuff. You can just iterate on the keys of your map, see if the attribute exists, and if it does assign it. Something along the lines of:
```js
for (const [attributeName, styleName] of attributesToStyle.entries()) {
    if (node.hasAttribute(attributeName) {
        node.style[styleName] = node.getAttribute(attributeName);
    }
}
```